### ISSUE TITLE
refactor `garden-debian-sources-list` script

### ISSUE THREAD

**Issue by gyptazy** [gardenlinux/gardenlinux#1433]
**What would you like to be added**:
While integrating a new native Debian mirror support for Garden Linux, we encountered several things that should be adjusted within the `garden-debian-sources-list` script. However, we probably should completely refactor this one after #1426 got merged.

There shouldn't be any `--gardenlinux-only` param anymore since this is our default. Next, we probably can remove older references of `squeeze` etc. With #1426 the orchestrating scripts are already adjusted and the handling of a non existent `--gardenlinux-only` param integrated. Therefore, we only may call it with an additional `--debian-mirror` param which adds the native Debian repository.

**Comment by gyptazy** [comment]
#1426 is currently blocked by the broken Github tests. However, if #1426 is still not merged until the 1st of December, I may include a fix for this issue within #1426. If anyone else may start with this earlier - go ahead!

**Comment by gyptazy** [comment]
DEP task #1426 is fixed, now. In review by @Vincinator and ready for merge.

**Comment by nanory** [comment]
Quick update to this issue. We had a discussion with @gehoern about this and the following things should be kept in mind while refactoring this script:
* This script shouldn't be part of the build pipeline anymore but rather be included in the artifact itself.
  * Keep in mind, that we will still need it for bootstrapping Garden Linux, but it should also be included in a running instead in order to modify sources.
* It should be more intuitive, so that it's easy to modify the sources of the Garden Linux image during runtime.
* Maybe, there is already such a script in Debian. If so, we should re-use it and adjust it for Garden Linux then. Needs to be verified.
* It should be possible to add and define the following sources:
  * Only Garden Linux
  * Only Debian
     * Just Debian testing
     * Just Debian unstable
* It should be possible to remove sources easily without modifying the APT sources by ourselves.
* We may should keep the `/etc/apt/source.list` empty and just use the drop in directory `/etc/apt/source.list.d`
  * There could be files in the drop in directory like: `gardenlinux.list`, `debian-testing.list`, ...
  * Via a flag, one can also add source packages sources instead of just binary sources.
* This tool should also consider priorities (take a look how it is already done in the current script).
  * It may be useful to define Garden Linux to priority 990+ to ensure hot fixes are applied no matter what the target version ist.
  * Take a look into here: https://linux.die.net/man/5/apt_preferences
* We should check, if the script should use the Deb822-style format instead of the One-Line-Style Format for better machine-parsability.
